packages/repo-tools/src/commands/generate-catalog-info/generate-catalog-info.ts

```diff
@@ -165,8 +165,17 @@ async function fixCatalogInfoYaml(options: FixOptions) {
     relativePath('.', yamlPath),
   );
   const safeName = packageJson.name
-    .replace(/[^a-z0-9_\-\.]+/g, '-')
-    .replace(/^[^a-z0-9]|[^a-z0-9]$/g, '');
+    .replace(/^[^\w\s]|[^a-z0-9]$/g, '')
```

Comment:
may be time to make a dedicated function for this since it happens twice and is super complex - and add a test file